diff --git a/src/common.h b/src/common.h
index c91d349..a19ad3a 100644
--- a/src/common.h
+++ b/src/common.h
@@ -25,7 +25,7 @@
 #define LDCONFIG_ALT_PATH         "/sbin/ldconfig.real"
 
 #define LIB_DIR                   "/lib64"
-#define USR_BIN_DIR               "/usr/bin"
+#define USR_BIN_DIR               "/opt/bin"
 #define USR_LIB_DIR               "/usr/lib64"
 #define USR_LIB32_DIR             "/usr/lib32"
 #define USR_LIB32_ALT_DIR         "/usr/lib"
