diff --git a/src/py/libcamera/meson.build b/src/py/libcamera/meson.build
index af19ffdd..142a5009 100644
--- a/src/py/libcamera/meson.build
+++ b/src/py/libcamera/meson.build
@@ -9,8 +9,10 @@ endif
 
 pycamera_enabled = true
 
-pybind11_proj = subproject('pybind11')
-pybind11_dep = pybind11_proj.get_variable('pybind11_dep')
+#pybind11_proj = subproject('pybind11')
+#pybind11_dep = pybind11_proj.get_variable('pybind11_dep')
+pybind11_incdir = include_directories('/usr/lib/python3.9/site-packages/pybind11/include')
+pybind11_dep = declare_dependency(include_directories : pybind11_incdir)
 
 pycamera_sources = files([
     'py_camera_manager.cpp',
