diff --git a/src/assimp/build_vars.cmake b/src/assimp/build_vars.cmake
index 0ceff6a..89d5dbb 100644
--- a/src/assimp/build_vars.cmake
+++ b/src/assimp/build_vars.cmake
@@ -25,7 +25,7 @@ if (${vsgXchange_assimp})
         assimp/assimp.cpp
     )
     set(EXTRA_INCLUDES ${EXTRA_INCLUDES} ${assimp_INCLUDE_DIRS})
-    set(EXTRA_LIBRARIES ${EXTRA_LIBRARIES} assimp::assimp)
+    set(EXTRA_LIBRARIES ${EXTRA_LIBRARIES} ${assimp_LIBRARIES})
     if(NOT BUILD_SHARED_LIBS)
         set(FIND_DEPENDENCY ${FIND_DEPENDENCY} "find_dependency(assimp)")
     endif()
