diff --git a/src/assimp/build_vars.cmake b/src/assimp/build_vars.cmake
index 0ceff6a..a5ceab0 100644
--- a/src/assimp/build_vars.cmake
+++ b/src/assimp/build_vars.cmake
@@ -12,13 +12,13 @@ endif()
 
 if (${vsgXchange_assimp})
 
+    message("\nassimp_VERSION "${assimp_VERSION})
+
     string(REPLACE "." ";" ASSIMP_VERSION_LIST ${assimp_VERSION})
     list(GET ASSIMP_VERSION_LIST 0 ASSIMP_VERSION_MAJOR)
     list(GET ASSIMP_VERSION_LIST 1 ASSIMP_VERSION_MINOR)
     list(GET ASSIMP_VERSION_LIST 2 ASSIMP_VERSION_PATCH)
 
-    message("\nassimp_VERSION "${assimp_VERSION})
-
     set(EXTRA_DEFINES ${EXTRA_DEFINES} "ASSIMP_VERSION_MAJOR=${ASSIMP_VERSION_MAJOR}" "ASSIMP_VERSION_MINOR=${ASSIMP_VERSION_MINOR}" "ASSIMP_VERSION_PATCH=${ASSIMP_VERSION_PATCH}" )
 
     set(SOURCES ${SOURCES}
