diff -u -r -N a/Makefile b/Makefile
--- a/Makefile	2022-02-14 10:11:03.000000000 -0500
+++ b/Makefile	2022-04-03 07:37:22.005646494 -0400
@@ -29,18 +29,12 @@
 
 all:
 	$(MAKE) -C lib
-	$(MAKE) -C test
-	$(MAKE) -C perf
 
 clean:
 	$(MAKE) -C lib clean
-	$(MAKE) -C test clean
-	$(MAKE) -C perf clean
 
 style:
 	$(MAKE) -C lib style
-	$(MAKE) -C test style
-	$(MAKE) -C perf style
 
 install:
 	$(MAKE) -C lib install
