diff --git a/src/vsg/CMakeLists.txt b/src/vsg/CMakeLists.txt
index 6467a0bd..573666ed 100644
--- a/src/vsg/CMakeLists.txt
+++ b/src/vsg/CMakeLists.txt
@@ -220,20 +220,6 @@ set(LIBRARIES PUBLIC
         Threads::Threads
 )
 
-# Check for std::atomic
-if(NOT MSVC AND NOT ANDROID AND NOT APPLE)
-  set(HAVE_CXX_ATOMIC_WITHOUT_LIB FALSE)
-  check_cxx_source_compiles("
-    #include <atomic>
-    std::atomic a;
-    int main() { return a; }"
-  HAVE_CXX_ATOMIC_WITHOUT_LIB)
-  if(NOT HAVE_CXX_ATOMIC_WITHOUT_LIB)
-    find_library(CXX_ATOMIC_LIBRARIES NAMES atomic atomic.so.1 libatomic.so.1 REQUIRED)
-    list(APPEND LIBRARIES ${CXX_ATOMIC_LIBRARIES})
-  endif()
-endif()
-
 if (glslang_FOUND)
     set(LIBRARIES ${LIBRARIES} PUBLIC ${glslang_LIBRARIES})
 
