diff -u -r -N a/src/tools/analyzer/Makefile.am b/src/tools/analyzer/Makefile.am
--- a/src/tools/analyzer/Makefile.am	2023-06-23 09:26:13.000000000 -0400
+++ b/src/tools/analyzer/Makefile.am	2023-08-23 08:41:32.166510729 -0400
@@ -1,5 +1,7 @@
 sss_analyze_pythondir = $(libexecdir)/sssd
+modulesdir = $(pkgpythondir)/modules
 
+if BUILD_PYTHON_BINDINGS
 dist_sss_analyze_python_SCRIPTS = \
     sss_analyze \
     $(NULL)
@@ -20,3 +22,4 @@
     modules/__init__.py \
     modules/request.py \
     $(NULL)
+endif
