--- a/lnet/autoconf/lustre-lnet.m4	2024-11-18 12:46:15.000000000 -0500
+++ b/lnet/autoconf/lustre-lnet.m4	2024-12-09 06:11:19.902497564 -0500
@@ -294,7 +294,7 @@
 
 AS_IF([test $ENABLEO2IB != "no"], [
 	EXTRA_CHECK_INCLUDE="$EXTRA_OFED_CONFIG $EXTRA_OFED_INCLUDE"
-	if test ! $O2IBPATH -ef $LINUX_OBJ; then
+	if test ! $(dirname $O2IB_SYMVER) -ef $LINUX_OBJ; then
 		KBUILD_EXTRA_SYMBOLS="$KBUILD_EXTRA_SYMBOLS $O2IBPATH/Module.symvers"
 	fi
 
