diff '--color=auto' -u -r -N a/src/perftest_parameters.h b/src/perftest_parameters.h
--- a/src/perftest_parameters.h	2025-11-20 13:09:32.703250342 -0500
+++ b/src/perftest_parameters.h	2025-11-20 13:09:39.767182914 -0500
@@ -168,7 +168,7 @@
 #define MIN_SL	      (0)
 #define MAX_SL	      (15)
 #define MIN_GID_IX    (0)
-#define MAX_GID_IX    (64)
+#define MAX_GID_IX    (256)
 #define MIN_QP_NUM    (1)
 #define MAX_QP_NUM    (16384)
 #define MIN_QP_MCAST  (1)
